main.nf

```diff
@@ -165,60 +170,126 @@ process StringTie2 {
                 }
 
     input:
-    set val(name), file(bam), file(annot) from ch_txome_reconstruction
+    set val(name), file(bam), val(annot) from ch_txome_reconstruction
+    val transcriptquant from ch_transcriptquant
 
     output:
-    set val(name), file(bam), file("*.out.gtf") into ch_txome_feature_count
+    set val(name), file(bam) into ch_txome_feature_count
+    val annot into ch_annot
     file("*.version") into ch_stringtie_version
+    val "${params.outdir}/stringtie2" into ch_stringtie_outputs
+    file "*.out.gtf"
+
+    when:
+    transcriptquant == "stringtie"
 
     script:
     """
     stringtie -L -G $annot -o ${name}.out.gtf $bam
     stringtie --version &> stringtie.version
     """
 }
+ch_stringtie_outputs
+   .unique()
+   .set {ch_stringtie_dir}
+ch_annot
+   .unique()
+   .set{ch_annotation}
+
+process GffCompare {
+    publishDir "${params.outdir}/stringtie2", mode: 'copy',
+        saveAs: { filename ->
+                      if (!filename.endsWith(".version")) filename
+                }
+    input:
+    val stringtie_dir from ch_stringtie_dir
+    val annot from ch_annotation
+    val transcriptquant from ch_transcriptquant
 
-// Combine channels for feature count with gtf from transcriptome reconstruction
-ch_annot_feature_count
-    .concat(ch_txome_feature_count)
-    .set{ ch_feature_count }
+    output:
+    val "$stringtie_dir/merged.combined.gtf" into ch_merged_gtf
 
+    when:
+    transcriptquant == "stringtie"
 
-/*
- * STEP 3 - FeatureCounts
- */
- process FeatureCounts {
+    script:
+    """
+    ls -d -1 $PWD/$stringtie_dir/*.out.gtf > $PWD/$stringtie_dir/gtf_list.txt
+    echo "$annot" >> $PWD/$stringtie_dir/gtf_list.txt
+    gffcompare -i $PWD/$stringtie_dir/gtf_list.txt -o $PWD/$stringtie_dir/merged
+    gffcompare --version &> gffcompare.version
+    """
+}
+
+ch_txome_feature_count
+   .combine(ch_merged_gtf)
+   .set {ch_feature_count}
+
+process FeatureCounts {
      publishDir "${params.outdir}/featureCounts_transcript", mode: 'copy',
          saveAs: { filename ->
-                 if (!filename.endsWith(".version") && !filename.endsWith(".gene_counts.txt")) filename
-                 }
-     publishDir "${params.outdir}/featureCounts_gene", mode: 'copy',
-         saveAs: { filename ->
-                 if (!filename.endsWith(".version") && !filename.endsWith(".transcript_counts.txt")) filename
+                 if (!filename.endsWith(".version")) filename
                  }
 
      input:
-     set val(name), file(bam), file(annot) from ch_feature_count
+     set val(name), file(bam), val(annot) from ch_feature_count
+     val transcriptquant from ch_transcriptquant
 
      output:
      file("*.txt") into ch_counts
      file("*.version") into ch_feat_counts_version
-     val "result/featureCounts_gene" into ch_deseq_indir
-     val "result/featureCounts_transcript" into ch_dex_indir
+     val "$baseDir/results/featureCounts_transcript" into ch_deseq2_indir
```

Comment:
should it $params.outdir instead? same for the line below

Reply:
I have gotten problems using $params.outdir instead of $baseDir/results, which $params.outdir maps to $baseDir/work/<random_string>/.